pkg/console/util.go

```diff
@@ -499,3 +506,56 @@ func retryRemoteConfig(configURL string, g *gocui.Gui) (*config.HarvesterConfig,
 func harvesterInstalled() (bool, error) {
 	return false, nil
 }
+
+func createPartitionLayout(devPath string) (string, error) {
+	// TODO(john): Use the yip/schema to define the partition layout. This requires the newer yip
+	// version because we need "Path" field of "Device" struct.
+	yipConfig := map[string]interface{}{
+		"stages": map[string][]interface{}{
+			"partitioning": {
+				map[string]interface{}{
+					"name": "Part layout",
+					"layout": map[string]interface{}{
+						"device": map[string]string{
+							"path": devPath,
+						},
+						"add_partitions": []yipSchema.Partition{
+							{
+								FSLabel:    "COS_OEM",
+								PLabel:     "oem",
+								Size:       50,
+								FileSystem: "ext4",
+							},
+							{
+								FSLabel:    "COS_STATE",
+								PLabel:     "state",
+								Size:       15360,
+								FileSystem: "ext4",
+							},
+							{
+								FSLabel:    "COS_RECOVERY",
+								PLabel:     "recovery",
+								Size:       8192,
+								FileSystem: "ext4",
+							},
+							{
+								FSLabel:    "COS_PERSISTENT",
+								PLabel:     "persistent",
+								Size:       30720,
```

Comment:
30G is a bit tight. 
My near-fresh installation uses around 19G for rancher things (including preloading images).
The partition stores logs, crash dumps and we need to consider an upgrade loads new images and need some temporary space too.

Reply:
We could set up a minimum requirement for the partition and a percentage-based size. For example `50G or 40% of the whole disk)`

Reply:
The log actually won't take too many spaces, but for safety concerns, I think we can escalate the default system partition size to 50G as per our doc description the minimum disk capacity would be 120GB.

Also, if we can add percentage rate calculation would be useful, e.g the maximum system partition disk would be 100G and will be increased 10G per 100G.

Reply:
@guangbochen I added the size calculation.

Another question: It also checks if [the selected disk's size meets our minimum requirement](https://github.com/harvester/harvester-installer/pull/160/files#diff-7585f2fd8ff9666817a029d282094ddc9ae07863f3af56712c47bda13d3b39b9R573). If this is a reasonable validation, I think we should validate the disk size a bit earlier, i.e. the "Choose installation target" page of the Installer?
![image](https://user-images.githubusercontent.com/5047724/141234546-3e3efbfb-c5c2-44ff-9d30-3a7e980e3b0c.png)